### PR TITLE
Fix validator updates ondelegate

### PR DESCRIPTION
**Pull request by xlab**
Originally missed patch from the upstream https://github.com/cosmos/cosmos-sdk/pull/17436
Update integration tests as well.

### CODE REVIEW

**Review by kruspy — APPROVED** [review]

File: x/staking/keeper/msg_server.go [hunk 1]
```diff
@@ -226,15 +226,10 @@ func (k msgServer) Delegate(goCtx context.Context, msg *types.MsgDelegate) (*typ
...
 			return nil, err
 		}
-		if err := k.SafelyIncreaseValidatorLiquidShares(ctx, validator, shares); err != nil {
+		validator, err = k.SafelyIncreaseValidatorLiquidShares(ctx, valAddr, shares)
+		if err != nil {
 			return nil, err
 		}
-		// Note: this is required for downstream uses of the validator variable
-		// since the validator's liquid shares were updated above
-		validator, found = k.GetValidator(ctx, valAddr)
```
Comment (puneet2019) [inline x/staking/keeper/msg_server.go:234]: we are sure about deleting this? I think we should keep this one

Comment (xlab) [reply x/staking/keeper/msg_server.go:234]: I am having 1:1 to the upstream version that is in Gaia now

Comment (puneet2019) [reply x/staking/keeper/msg_server.go:234]: my bad, all good, `SafelyIncreaseValidatorLiquidShares` now returns updated validator too

Comment (xlab) [reply x/staking/keeper/msg_server.go:234]: `validator, err = k.SafelyIncreaseValidatorLiquidShares` updates it above, no?

**Review by puneet2019 — APPROVED** [review]

**Pull request merged**